cluster/addons/fluentd-gcp/event-exporter.yaml

```diff
@@ -29,10 +29,11 @@ subjects:
 apiVersion: apps/v1beta1
 kind: Deployment
 metadata:
-  name: event-exporter-v0.1.4
```

Comment:
AFAIU yes, the old deployment will be deleted: https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/addon-manager

> Addon will be deleted when its manifest file is deleted.

I think changing the name of the deployment is the same as removing the old addon and creating a new one